File: src/javascript5/checks.js

(function(exports, imports) {
    //::imports

    //::module checks
    var checks = exports.Checks = {
        //::function invalid
        invalid: function(error) {
            console.error(error);
        },

        //::function valid
        valid: function() {
            //do nothing
        },

        //::function isDefined
        //@@depends valid, invalid
        isDefined: function(val, name) {
            return val === undefined ? checks.invalid(name + " is undefined") : checks.valid();
        },

        //::function isPositive
        //@@depends valid, invalid
        isPositive: function(nr) {
            return nr < 0 ? checks.invalid("negative: " + nr) : checks.valid();
        },

        //::function isNegative
        //@@depends valid, invalid
        isNegative: function(nr) {
            return nr > 0 ? checks.invalid("positive: " + nr) : checks.valid();
        },

        //::addbefore
        stub: null
    }
})(window);

File: src/ruby2/checks.rb
#::module checks
module Checks
  extend Checks
#::imports

  #::function valid
  def valid()
    true
  end

  #::function invalid
  def invalid(errorMsg)
    errorMsg
  end

  #::function isDefined
  #@@depends valid,invalid
  def is_defined(val, name)
    val != nil ? valid() : invalid("undefined: #{name}")
  end

  #::function isPositive
  #@@depends valid,invalid
  def is_positive(val)
    val > 0 ? valid() : invalid("negative: #{val}")
  end

  #::function isNegative
  #@@depends valid,invalid
  def is_negative(val)
    val < 0 ? valid() : invalid("positive: #{val}")
  end
  #::addbefore
end